platforms/documentation/docs/src/docs/userguide/authoring-builds/build_environment.adoc

```diff
@@ -131,54 +129,82 @@ spring-boot.version = 2.2.1.RELEASE
 [[sec:project_properties]]
 == Project properties
 
-Project properties are specific to your Gradle project, they are variables or configuration blocks defined in your build script or in the `project` object.
+Project properties are specific to your Gradle project, they are variables or configuration blocks defined in your build script or in the `Project` object.
+Project properties can be retrieved:
 
-For example, the project property `dryRun`, in the build file below, enables a specific test mode:
+- Lazily: `providers.gradleProperty()`
+- Eagerly: `project.ext()`, `project.findProperty()`, `project.property()`, `project.('<propertyName>')`
 
-.build.gradle
-[source,groovy]
-----
-test {
-  predictiveSelection {
-    enabled = true
-      dryRun = true
-  }
-}
-----
+If a project property is referenced but does not exist, an exception may be thrown and the build will fail.
+You should check for the existence of a project property before directly accessing it with `project.('<propertyName>')` by using the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:hasProperty(java.lang.String)[`project.hasProperty()`] method.
```

Comment:
If the `providers.gradleProperty()` method is the preferred way to access properties, then you can do all this using the `Provider` APIs. `providers.gradleProperty()` won't throw an exception if the property doesn't exist, only if `.get()` is called when there is no value present, but you can check for presence by using `.isPresent`.

Reply:
I went for a 180 and got too specific. Removing this,